From 52b5fed00d383ce27f54af12cb1fe2134017af20 Mon Sep 17 00:00:00 2001
From: xtex <xtex@astrafall.org>
Date: Sun, 8 Mar 2026 11:13:36 +0800
Subject: [PATCH 132/133] AOSC OS: add search-prefix option
X-Developer-Signature: v=1; a=openpgp-sha256; l=1313; i=xtex@astrafall.org;
 h=from:subject; bh=Wg/QGU/y3SqGK066yJ6iklKZ673kaZSm9H7/pX51dAw=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78gR5viW8Yj3N4sL5VfHfQrOgrS8msgrKsW940bHj4
 MpXzXdbOyayMIhxMViKKbIUGTZ4s+qk84suK5eFmcPKBDJEWqSBgYGBgYWBLzcxr9RIx0jPVNtQ
 z9BQx0jHiIGLUwCmOj2fkWHphHefTB8u1VnjcpfXVCzDoSora07psl/Zvz4t3N+4sL2b4Z8hf1S
 6+pm+Ld7Ssqo7jvovjYrY/8BiytfDlh361dOlJnEDAA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

Change-Id: I2bcee6404329238bf99b240057cbe12c9daf8f1c
---
 build.zig | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/build.zig b/build.zig
index b8f9c62c7dbf..0cd99a443f24 100644
--- a/build.zig
+++ b/build.zig
@@ -104,6 +104,7 @@ pub fn build(b: *std.Build) !void {
 
     const static_llvm = b.option(bool, "static-llvm", "Disable integration with system-installed LLVM, Clang, LLD, and libc++") orelse false;
     const enable_llvm = b.option(bool, "enable-llvm", "Build self-hosted compiler with LLVM backend enabled") orelse static_llvm;
+    const search_prefix = b.option([][]const u8, "search-prefix", "AOSC: Add custom search prefix") orelse &.{};
     const llvm_has_m68k = b.option(
         bool,
         "llvm-has-m68k",
@@ -303,6 +304,11 @@ pub fn build(b: *std.Build) !void {
     const version = try b.allocator.dupeZ(u8, version_slice);
     exe_options.addOption([:0]const u8, "version", version);
 
+    for (search_prefix) |path| {
+        exe.root_module.addLibraryPath(.{ .cwd_relative = path });
+        exe.root_module.addSystemIncludePath(.{ .cwd_relative = path });
+    }
+
     if (enable_llvm) {
         const cmake_cfg = if (static_llvm) null else blk: {
             if (findConfigH(b, config_h_path_option)) |config_h_path| {
-- 
2.52.0

